### ISSUE TITLE
[PROJECT ONBOARDING] Keycloak

### ISSUE THREAD

**Issue by amye** [cncf/sandbox#161]
Welcome to CNCF Project Onboarding!
This is an issue created to help onboard your project into the CNCF after the TOC has voted to accept your project.
We would like to complete onboarding within one month of acceptance.

From the project side, please ensure that you:

- [x] Understand the project proposal process and reqs: <https://github.com/cncf/toc/blob/main/process/project_proposals.md#introduction>
- [x] Understand the services available for your project at CNCF <https://www.cncf.io/services-for-projects/>
- [x] Ensure your project meets the CNCF IP Policy: <https://github.com/cncf/foundation/blob/master/charter.md#11-ip-policy>
- [x] Review the online programs guidelines: <https://github.com/cncf/foundation/blob/master/online-programs-guidelines.md>
- [x] Understand the trademark guidelines: <https://www.linuxfoundation.org/en/trademark-usage/>
- [x] Understand the license allowlist: <https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md#approved-licenses-for-allowlist>
- [x] Is your project working on written, open governance? see <https://contribute.cncf.io/maintainers/governance/>
- [x] Slack: Are your slack channels migrated to the Kubernetes or CNCF Slack? (see <https://slack.com/help/articles/217872578-Import-data-from-one-Slack-workspace-to-another> for more details)
- [x] Is your project in its own separate neutral github organization?
- [x] Submitted a Pull request to add your project as a incubating project to <https://landscape.cncf.io>
- [x] Create maintainer list + add to aggregated <https://maintainers.cncf.io> list by submitting a PR to it
- [x] Artwork: Submit a pull request to <https://github.com/cncf/artwork> with your artwork
- [x] Domain: transfer domain to the CNCF - <https://jira.linuxfoundation.org/plugins/servlet/theme/portal/2/create/63>

Things that CNCF will need from the project:

- [x] Provide emails for the maintainers added to <https://maintainers.cncf.io> in order to get access to the maintainers mailing list and ServiceDesk - project-onboarding@cncf.io is the best email to send those to
- [x] Trademarks: transfer any trademark and logo mark assets over to the LF - <https://github.com/cncf/foundation/tree/master/agreements> has agreements
- [x] GitHub: ensure 'thelinuxfoundation' and 'caniszczyk' are added as initial org owners, this helps us make sure we have continuity of GH ownership that we will onboard to our GitHub Enterprise instance: https://github.com/enterprises/cncf
- [x] GitHub: ensure [DCO](https://github.com/apps/dco) or [CLA](https://github.com/cncf/cla) are enabled for all GitHub repositories of the project
- [x] GitHub: ensure that that the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md) (or your adopted version of it) are explicitly referenced at the project's README on GitHub
- [x] Website: ensure LF footer is there and [website guidelines](https://github.com/cncf/foundation/blob/master/website-guidelines.md) followed (if your project doesn't have a dedicated website, please adopt those guidelines to the README file of your project on GitHub).
- [x] Website: Analytics transferred to projects@cncf.io
- [x] OpenSSF Best Practices Badge: Start on an OpenSSF Best Practices Badge <https://bestpractices.coreinfrastructure.org/en>

Things that the CNCF will do or help the project to do:

- [x] Devstats: add to devstats <https://devstats.cncf.io/>
- [x] Insights: add to LFX Insights <https://insights.v3.lfx.linuxfoundation.org/>
- [x] Marketing: update relevant intro + slide decks
- [x] Events: update CFP + Registration + CFP Area forms
- [x] ServiceDesk: confirm maintainers have read <https://www.cncf.io/services-for-projects/>
- [x] CNCF Welcome Email Sent to confirm maintainer list access
- [x] Book time with the team with <http://project-meetings.cncf.io>
- [x] Create space for meetings/events on <https://community.cncf.io>, e.g., <https://community.cncf.io/pravega-community/> - (<https://github.com/cncf/communitygroups/blob/main/README.md#cncf-projects>)
- [x] Adopt a license scanning tool, like [FOSSA](https://fossa.com/) or [Snyk](https://snyk.io/)
- [x] Help add the project to the CNCF store

**Comment by abstractj** [comment]
@mrbobbytables why this was moved under sandbox, if the project is under incubation? cc @stianst

**Comment by stianst** [comment]
@caniszczyk @jeefy `Domain: transfer domain to the CNCF` has been completed

**Comment by stianst** [comment]
@mrbobbytables Why was this moved to Sandbox? cc: @caniszczyk @jeefy

**Comment by krook** [comment]
> [@caniszczyk](https://github.com/caniszczyk) [@jeefy](https://github.com/jeefy) `Domain: transfer domain to the CNCF` has been completed

Excellent, I checked that off the list.

So the final remaining items here relate to GitHub @abstractj @stianst: 

> - [ ] GitHub: ensure 'thelinuxfoundation' and 'caniszczyk' are added as initial org owners, this helps us make sure we have continuity of GH ownership that we will onboard to our GitHub Enterprise instance: https://github.com/enterprises/cncf

For reasons we've discussed earlier around GitHub Actions, we **won't** move the organization into GitHub Enterprise yet. However, in order to  maintain a neutral home for the project, we'll still need to add [`thelinuxfoundation`](https://github.com/thelinuxfoundation) as an organization owner. 

> - [ ] Insights: add to LFX Insights https://insights.v3.lfx.linuxfoundation.org/

When that's done, we'll also then be able use that id to add the GitHub app to enable Keycloak for onboarding to LFX Insights. At that point we can call onboarding (as an Incubating project) complete.

**Comment by abstractj** [comment]
Thanks, @krook at the moment we are blocked waiting for a solution to scaling GitHub Action runners for us to have confidence that CNCF will cover our needs. According to [Hosted tools/resources](https://contribute.cncf.io/resources/project-services/hosted-tools/), which is part of what we were looking at before donating Keycloak to the CNCF; it clearly states: The CNCF covers CI requires for our hosted projects and allows those projects to select their platforms.

Our CI needs are GitHub Actions with standard GitHub runners, and we have a need for a minimum of 50 parallel jobs, but that’s an absolute minimum.

Robert Kielty mentioned that you should be able to look into it on your return, if I understood it correcly. As soon as we have those things sorted out, we should be able to proceed with the next steps. cc @stianst

**Comment by idvoretskyi** [comment]
It’s on us, apologies for the delay. We’ll keep you posted on the CI
progress.

Ihor Dvoretskyi ***@***.***>
Sr. Developer Advocate, Cloud Native Computing Foundation https://cncf.io/

**Comment by krook** [comment]
@abstractj agreed that the CI/GitHub Actions is a blocker for bringing the Keycloak org into the GitHub Enterprise account. Work on a solution there is still underway to support what you need, which is why we **won't** move the organization into GitHub Enterprise yet.

But, we can still address the two open checkmarks in this checklist in the meantime by adding `thelinuxfoundation` as an organization owner on your existing organization and therefore being able to get you into LFX Insights.

**Comment by stianst** [comment]
I will not proceed with adding `thelinuxfoundation` as an organization owner of the Keycloak GitHub organization until the issue around CI/CD and payment for the organization has been resolved.

Quote from https://contribute.cncf.io/resources/project-services/hosted-tools/:

_The CNCF covers CI needs for our hosted projects and allows those projects to select their own platforms_

Our needs are a minimum of 50 parallell **GitHub-hosted runners**, ideally it would be 100+ to prevent larger queues. We have attempted to use the self-hosted runners from the CNCF, but these are not suitable for our purposes and not delivered with the level of service we need to prevent disruptions to contributors and releases.

As long as the CNCF does not fill its promise of covering our CI needs we are unable to proceed with the onboarding.

**Comment by stianst** [comment]
We are looking forward to transfering the Keycloak organization to CNCF, or at the least add `thelinuxfoundation` as an organization owner. However, right now due to how the billing is handled I'm not able to do so. I'm soon going on PTO for Xmas, but will schedule a call with you in January so we can figure out the next steps and complete the onboarding.

**Comment by krook** [comment]
Now that Keycloak is onboarded to Insights, we can mark this onboarding issue complete 🎉 

We can now work on GHE/Actions and focus on those Service Desk tickets.
